desc/compute/_surface.py

```diff
@@ -427,6 +427,173 @@ def _e_zeta_z_FourierRZToroidalSurface(params, transforms, profiles, data, **kwa
     return data
 
 
+@register_compute_fun(
+    name="Phi",
+    label="\\Phi",
+    units="A",
+    units_long="amperes",
+    description="Surface current potential",
+    dim=2,
+    params=["I", "G", "Phi_mn"],
+    transforms={"Phi": [[0, 0, 0]]},
+    profiles=[],
+    coordinates="tz",
+    data=[],
+    parameterization="desc.magnetic_fields.FourierCurrentPotentialField",
+    basis="basis",
+)
+def _Phi_FourierCurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi"] = (
+        transforms["Phi"].transform(params["Phi_mn"])
+        + params["G"] * transforms["Phi"].nodes[:, 2].flatten(order="F") / 2 / jnp.pi
+        + params["I"] * transforms["Phi"].nodes[:, 1].flatten(order="F") / 2 / jnp.pi
+    )
+    return data
+
+
+@register_compute_fun(
+    name="Phi_t",
+    label="\\partial_{\\theta}Phi",
+    units="A",
+    units_long="amperes",
+    description="Surface current potential, poloidal derivative",
+    dim=2,
+    params=["I", "G", "Phi_mn"],
+    transforms={"Phi": [[0, 1, 0]]},
+    profiles=[],
+    coordinates="tz",
+    data=[],
+    parameterization="desc.magnetic_fields.FourierCurrentPotentialField",
+    basis="basis",
+)
+def _Phi_t_FourierCurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi_t"] = (
+        transforms["Phi"].transform(params["Phi_mn"], dt=1) + params["I"] / 2 / jnp.pi
+    )
+    return data
+
+
+@register_compute_fun(
+    name="Phi_z",
+    label="\\partial_{\\zeta}Phi",
+    units="A",
+    units_long="amperes",
+    description="Surface current potential, toroidal derivative",
+    dim=2,
+    params=["I", "G", "Phi_mn"],
+    transforms={"Phi": [[0, 0, 1]]},
+    profiles=[],
+    coordinates="tz",
+    data=[],
+    parameterization="desc.magnetic_fields.FourierCurrentPotentialField",
+    basis="basis",
+)
+def _Phi_z_FourierCurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi_z"] = (
+        transforms["Phi"].transform(params["Phi_mn"], dz=1) + params["G"] / 2 / jnp.pi
+    )
+    return data
+
+
+@register_compute_fun(
+    name="Phi",
+    label="\\Phi",
+    units="A",
+    units_long="amperes",
+    description="Surface current potential",
+    dim=2,
+    params=["params"],
+    transforms={"grid": [], "potential": []},
+    profiles=[],
+    coordinates="tz",
+    data=[],
+    parameterization="desc.magnetic_fields.CurrentPotentialField",
+    basis="basis",
+)
+def _Phi_CurrentPotentialField(params, transforms, profiles, data, **kwargs):
+    data["Phi"] = transforms["potential"](
+        transforms["grid"].nodes[:, 1],
+        transforms["grid"].nodes[:, 2],
+        **params["params"][0]
```

Comment:
this `[0]` is here becauuse the params["params"] is an `ndarray` type, not the `dict` type I expected, given that the `CurrentPotentialField.params` attribute is a dictionary. I was unsure of a better way to account for the fact that this class allows arbitrary functions for the potential and thus we don't know what the params will be beforehand, so they are not each individually attributes of the class.

Reply:
were you able to figure out why its an array of dict instead of just a dict? AFAIK we don't do any weird wrapping or casting

Reply:
https://github.com/PlasmaControl/DESC/blob/7fcfdd3bbb4964440ce8460d0083374c331c72e7/desc/compute/utils.py#L292

this line turns anything in the params dictionary into a 1D ndarray

Reply:
I could avoid this by grabbing the params dict through the `get_transforms` function instead, but that is kinda confusing. This seems to work for now

Reply:
hmm ok that might break things if you try to jit or optimize the params of a current potential field though. np arrays get converted to jax arrays but jax arrays can't hold arbitrary objects like dictionaries.

we should probably change that to not cast dicts to arrays, and maybe use `jnp.atleast_1d` to make sure we're not doing anything funky that jax doesn't like

Reply:
you probably dont want to go through `get_transforms` because that would make it a constant during any optimization

Reply:
I think we want something like this:
```python
temp_params = {}
for name in params:
    p = getattr(obj, name)
    if isinstance(p, dict):
        temp_params[name] = p
    else:
        temp_params[name] = jnp.atleast_1d(p)
params = temp_params